SMPP/SmppClient/ESMEConnection.cs

```diff
@@ -25,7 +25,7 @@ internal class ESMEConnection : IDisposable
         /// <param name="phoneNumber"></param>
         /// <param name="dataCoding"></param>
         /// <param name="message"></param>
-        public delegate void RECEIVED_MESSAGE_HANDLER(string logKey, string serviceType, Ton sourceTon, Npi sourceNpi, string shortLongCode, DateTime dateReceived, string phoneNumber, DataCodings dataCoding, string message);
+        public delegate void RECEIVED_MESSAGE_HANDLER(string logKey, MessageTypes messageType, string serviceType, Ton sourceTon, Npi sourceNpi, string shortLongCode, DateTime dateReceived, string phoneNumber, DataCodings dataCoding, string message);
```

Comment:
Before we break the API I have a question.

Is this change to enhance logging or to fix broken logic?

Reply:
This is to separate the incoming message and delivery receipt.